src/calendar/day/basic/index.tsx

```diff
@@ -28,214 +26,188 @@ export interface BasicDayProps {
   disableAllTouchEventsForDisabledDays?: boolean;
   /** Disable all touch events for inactive days. can be override with disableTouchEvent in markedDates*/
   disableAllTouchEventsForInactiveDays?: boolean;
-  
-  /** Test ID*/
+
+  /** Test ID */
   testID?: string;
   /** Accessibility label */
   accessibilityLabel?: string;
 }
 
-export default class BasicDay extends Component<BasicDayProps> {
-  static displayName = 'BasicDay';
-
-  static propTypes = {
-    state: PropTypes.oneOf(['selected', 'disabled', 'inactive', 'today', '']),
-    marking: PropTypes.any,
-    markingType: PropTypes.oneOf(values(Marking.markings)),
-    theme: PropTypes.object,
-    onPress: PropTypes.func,
-    onLongPress: PropTypes.func,
-    date: PropTypes.object,
-    disableAllTouchEventsForDisabledDays: PropTypes.bool,
-    disableAllTouchEventsForInactiveDays: PropTypes.bool
-
-  };
-
-  style = styleConstructor(this.props.theme);
-
-  shouldComponentUpdate(nextProps: BasicDayProps) {
-    return shouldUpdate(this.props, nextProps, [
-      'children',
-      'state',
-      'markingType',
-      'marking',
-      'onPress',
-      'onLongPress',
-      'date'
-    ]);
-  }
-
-  onPress = () => {
-    this.props.onPress?.(this.props.date);
-  };
-
-  onLongPress = () => {
-    this.props.onLongPress?.(this.props.date);
-  };
-
-  get marking() {
-    return this.props.marking || {};
-  }
-
-  shouldDisableTouchEvent() {
-    const {disableAllTouchEventsForDisabledDays, disableAllTouchEventsForInactiveDays} = this.props;
-    const {disableTouchEvent} = this.marking;
+const BasicDay = (props: BasicDayProps) => {
+  const {
+    theme,
+    date,
+    onPress,
+    onLongPress,
+    markingType,
+    marking,
+    state,
+    disableAllTouchEventsForDisabledDays,
+    disableAllTouchEventsForInactiveDays,
+    accessibilityLabel,
+    children,
+    testID
+  } = props;
+  const style = styleConstructor(theme);
```

Comment:
Keep style as a ref, otherwise you are constructing it on every render (and creating a new style object ref)

Reply:
Thanks